Error in setting default value for option types to `none` in nested struct

### Describe the bug

- Setting default value of option type to `none` causes error. Anything else runs successfully.
- Also it prints incorrectly
  ```v
  d := Data{} // see reproduction steps for definition
  println(d)
  println(d.a)
  ```
  gives the output
  ```
  Data{
      a: 0
      b: 1
      c: 0
  }
  Option(error: none)
  ```
  I think it should be 
  ```
  Data{
      a: none
      b: 1
      c: none
  }
  Option(none)
  ```


### Expected Behavior

Should run successfully with a warning.

### Current Behavior

```
examples/check_encode.v:40:11: warning: unnecessary default value of `none`: struct fields are zeroed by default
   38 |     a ?int
   39 |     b ?int = 1
   40 |     c ?int = none
      |              ~~~~
   41 | }
   42 |
==================
C:/Users/dhrit/AppData/Local/Temp/v_0/check_encode.2632598036814096107.tmp.c:7630: warning: cast between pointer and integer of different size
C:/Users/dhrit/AppData/Local/Temp/v_0/check_encode.2632598036814096107.tmp.c:12490: error: field expected
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang 
```

### Reproduction Steps

```v
module main

struct Data {
	a ?int
	b ?int = 1
	c ?int = none // error
}

struct Data2 {
	d Data
}

fn main() {
	d := Data2{}
	println(d)
}

```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.3 9c511e0

### Environment details (OS name and version, etc.)

```
OS: windows, Microsoft Windows 11 Home Single Language v22621 64-bit
Processor: 8 cpus, 64bit, little endian, 
CC version: Error: exec failed (CreateProcess) with code 2: The system cannot find the file specified.
 cmd: cc --version

getwd: C:\Users\dhrit\.vmodules\vbson
vmodules: C:\Users\dhrit\.vmodules
vroot: C:\BIN\v
vexe: C:\BIN\v\v.exe
vexe mtime: 2023-02-26 08:44:50
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.3 9794a23.9c511e0

Git version: git version 2.35.1.windows.2
Git vroot status: weekly.2023.07-53-g9c511e03
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 1e6e7c6f
```

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -306,13 +306,28 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 
 fn (mut g Gen) zero_struct_field(field ast.StructField) bool {
 	sym := g.table.sym(field.typ)
-	if sym.kind == .struct_ {
-		info := sym.info as ast.Struct
-		if info.fields.len == 0 {
+	field_name := if sym.language == .v { c_name(field.name) } else { field.name }
+	if sym.info is ast.Struct {
+		if sym.info.fields.len == 0 {
 			return false
+		} else if !field.has_default_expr {
+			mut has_option_field := false
+			for fd in sym.info.fields {
+				if fd.typ.has_flag(.option) {
+					has_option_field = true
+					break
+				}
+			}
+			if has_option_field {
+				default_init := ast.StructInit{
+					typ: field.typ
+				}
+				g.write('.${field_name} = ')
+				g.struct_init(default_init)
+				return true
+			}
 		}
 	}
-	field_name := if sym.language == .v { c_name(field.name) } else { field.name }
 	g.write('.${field_name} = ')
 	if field.has_default_expr {
 		if sym.kind in [.sum_type, .interface_] {

--- a/vlib/v/tests/nested_option_struct_init_test.v
+++ b/vlib/v/tests/nested_option_struct_init_test.v
@@ -1,0 +1,18 @@
+struct Data {
+	a ?int
+	b ?int = 1
+	c ?int = none
+}
+
+struct Data2 {
+	d Data
+}
+
+fn test_nested_option_struct_init() {
+	d2 := Data2{}
+	println(d2)
+	assert d2.d.a == none
+	assert d2.d.b != none
+	assert d2.d.b? == 1
+	assert d2.d.c == none
+}

--- a/vlib/v/tests/nested_option_struct_init_test.v
+++ b/vlib/v/tests/nested_option_struct_init_test.v
@@ -1,7 +1,6 @@
 struct Data {
 	a ?int
 	b ?int = 1
-	c ?int = none
 }
 
 struct Data2 {
@@ -14,5 +13,4 @@ fn test_nested_option_struct_init() {
 	assert d2.d.a == none
 	assert d2.d.b != none
 	assert d2.d.b? == 1
-	assert d2.d.c == none
 }

